web/src/main/scala/net/wiringbits/components/widgets/ResendVerifyEmailForm.scala

```diff
@@ -0,0 +1,130 @@
+package net.wiringbits.components.widgets
+
+import com.alexitc.materialui.facade.materialUiCore.{components => mui, materialUiCoreStrings => muiStrings}
+import com.alexitc.materialui.facade.materialUiCore.mod.PropTypes.Color
+import com.alexitc.materialui.facade.react.components.Fragment
+import net.wiringbits.AppContext
+import net.wiringbits.core.I18nHooks
+import net.wiringbits.forms.ResendVerifyEmailFormData
+import net.wiringbits.ui.components.inputs.EmailInput
+import net.wiringbits.webapp.utils.slinkyUtils.components.core.ErrorLabel
+import net.wiringbits.webapp.utils.slinkyUtils.components.core.widgets.{CircularLoader, Container, Title}
+import net.wiringbits.webapp.utils.slinkyUtils.components.core.widgets.Container.{Alignment, EdgeInsets}
+import net.wiringbits.webapp.utils.slinkyUtils.forms.StatefulFormData
+import org.scalajs.dom
+import org.scalajs.macrotaskexecutor.MacrotaskExecutor.Implicits.global
+import slinky.core.{FunctionalComponent, SyntheticEvent}
+import slinky.core.annotations.react
+import slinky.core.facade.Hooks
+import slinky.web.html._
+
+import scala.util.{Failure, Success}
+
+@react object ResendVerifyEmailForm {
+  case class Props(ctx: AppContext)
+
+  val component: FunctionalComponent[Props] = FunctionalComponent[Props] { props =>
+    val texts = I18nHooks.useMessages(props.ctx.$lang)
+    val (formData, setFormData) = Hooks.useState(
+      StatefulFormData(
+        ResendVerifyEmailFormData.initial(
+          emailLabel = texts.email
+        )
+      )
+    )
+
+    def onDataChanged(f: ResendVerifyEmailFormData => ResendVerifyEmailFormData): Unit = {
+      setFormData { current =>
+        current.filling.copy(data = f(current.data))
+      }
+    }
+
+    def handleSubmit(e: SyntheticEvent[_, dom.Event]): Unit = {
+      e.preventDefault()
+
+      if (formData.isSubmitButtonEnabled) {
+        setFormData(_.submit)
+        for {
+          request <- formData.data.submitRequest
+            .orElse {
+              setFormData(_.submissionFailed(texts.completeData))
+              None
+            }
+        } yield props.ctx.api.client
+          .sendEmailVerificationToken(request)
+          .onComplete {
+            case Success(_) =>
+              setFormData(_.submitted)
```

Comment:
We will likely need to show something to the user if the request succeed (a dialog, a text, send him into a new page, whatever)